app.py

```diff
@@ -472,6 +509,49 @@ def invoice():
     #TODO: other statuses..
     return render_template("invoice.html", invoice=invoice, order=order, error=error, qrcode_svg=qrcode_svg, url=url)
 
+@app.route('/bronze_oauth')
+def bronze_oauth():
```

Comment:
this function seems to duplicate a lot from utilities()?

Reply:
im not seeing the duplication:
```
@app.route("/utilities")
def utilities():
    utilities = Utility.all_alphabetical(db.session)
    url = app.config["BRONZE_ADDRESS"] + '/Manage/Kyc'
    if not bronze.authorized:
        flash('Click "ALLOW" to proceed, once login', 'error')
        return render_template("utilities.html", utilities=utilities, url=url)
    else:
        return redirect(url_for('bronze_oauth'))
```